npm/generate_docs.js

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+// @ts-check
+
+import { existsSync, mkdirSync, writeFileSync } from "node:fs";
```

Comment:
See prior discussion. At some point we may want to make greater use of this API for an in-editor 'cargo docs' like feature. In editor (and in Wasm in general) you can't write to the file system. I don't see building a new executable just for generating docs as being any less complex than 56 lines of JavaScript.

Reply:
> I don't see building a new executable just for generating docs as being any less complex than 56 lines of JavaScript.

While I'm 100% on board with the wasm approach, I feel like you're overstating the complexity of adding a binary in Rust. It's like 10 lines of code - you add a main() method and a couple of lines in cargo.toml.

I do understand we need the WASM API for different reasons though, not because it's less complex.

Reply:
I agree. I'm not overstating it. I get that's trivial _also_. Just trying to make the point this is trivial, and not "unnecessary complexity" per the original comment.